text/0039-sdks-report-file-IO-on-main-thread.md

```diff
@@ -0,0 +1,69 @@
+* Start Date: 2022-11-28
+* RFC Type: decision
+* RFC PR: [#33](https://github.com/getsentry/rfcs/pull/33)
+* RFC Status: draft
+* RFC Driver: [Philipp Hofmann](https://github.com/philipphofmann)
+
+# Summary
+
+This RFC aims to clarify if SDKs should report file I/O on the main thread (__FIOMT__) as errors or performance issues.
+
+# Motivation
+
+On June 21, 2022, we decided with [DACI](https://www.notion.so/sentry/Performance-Issue-Creation-POC-e521772ebccb482b83b08f4f8a3db2cb) to create performance issues in Ingest. While implementing the FIOMT for Android, the question arose as to why SDKs don't report FIOMT.
+
+# Options Considered
+
+## Option 1: SDKs report FIOMT as errors
+
+SDKs report FIOMT as errors with a stacktrace. 
+
+To clarify the threshold and configuration, an experimental feature phase can help to get feedback.
+
+### Pros
+
+1. SDKs can capture a stack trace which will helps actionability and fingerprinting/grouping.
+2. Users don't have to enable performance monitoring.
+3. No running transaction required.
+4. Sentry can tie together transaction and error far more easily since both objects exist at the point of time the performance issue will be created.
+
+### Cons
+
+1. Double dipping quotas, sending the transaction and the error created within that transaction.
+
+Including [cons 1-3 of option 2](#option-2-cons).
+
+## Option 2: SDKs report FIOMT as performance issues
+
+SDKs detect and report FIOMT as a performance issue. To achieve this need to:
+
+1. answer billing questions.
+2. make changes in Ingest to accept performance issues.
+
+### Pros
+
+1. No double dipping of quotas.
```

Comment:
2. FIOMT is correctly categorized as a Performance Issue and returned when searching inside Sentry for performance issues
2. FIOMT's issue details are presented as a Performance Issue, highlighting root causes and resources to fix the issue
2. Performance Issue-specific quotas and thresholds apply to FIOMT
2. Planned UX and workflow changes specific to performance issues also apply to FIOMT issues

With the opposite of all these things being cons of option 1 😄

Reply:
Thank you @mjq-sentry 💯